src/Propel/Runtime/ActiveQuery/ModelCriteria.php

```diff
@@ -1044,17 +1052,32 @@ public function addSelectQuery(Criteria $subQueryCriteria, $alias = null, $addAl
      *
      * @param bool $force To enforce adding columns for changed alias, set it to true (f.e. with sub selects)
      *
-     * @return $this The current object, for fluid interface
+     * @return $this|\Propel\Runtime\ActiveQuery\ModelCriteria
      */
     public function addSelfSelectColumns($force = false)
     {
         if ($this->isSelfSelected && !$force) {
             return $this;
         }
 
-        /** @var string $tableMap */
-        $tableMap = $this->modelTableMapName;
-        $tableMap::addSelectColumns($this, $this->useAliasInSQL ? $this->modelAlias : null);
+        /** @var string $tableMapClassName */
+        $tableMapClassName = $this->modelTableMapName;
+        $alias = ($this->useAliasInSQL) ? $this->modelAlias : null;
+
+        return $this->addSelfSelectColumnsFromTableMapClass($tableMapClassName, $alias);
+    }
+
+    /**
+     * Adds the select columns for the given table.
+     *
+     * @param string $tableMapClassName
```

Comment:
I'm not privy to the coding standards in this project, but technically these `@param` annotations are redundant with the (stronger, therefore better :+1: ) native-PHP typehints in the function signature.